src/libs/hooks/useTheme.ts

```diff
@@ -0,0 +1,49 @@
+import { useEffect, useState } from "react";
+
+export const useTheme = (currentTheme: string) => {
+  const [darkTheme, setDarkTheme] = useState<boolean>(currentTheme === "dark");
+
+  const themeToggleHandler = async () => {
+    const newTheme = !darkTheme;
+    let theme;
+
+    if (newTheme) {
+      theme = "dark";
+    } else {
+      theme = "light";
+    }
+
+    setDarkTheme(newTheme); // 상태 변경
+
+    await fetch("/api/theme", {
+      method: "POST",
+      body: JSON.stringify({ theme }),
+      headers: new Headers({
+        "Content-Type": "application/json",
+      }),
+    });
+  };
+
+  useEffect(() => {
+    if (darkTheme !== undefined) {
+      if (darkTheme) {
+        document.body.setAttribute("data-theme", "dark");
+      } else {
+        document.body.removeAttribute("data-theme");
+      }
+    }
```

Comment:
전반적으로 조건문 스코프를 조금 더 효율적으로 핸들링하면 좋을 것 같다는 생각이 듭니다.
위 조건문도 잘 이해가 되지 않거든요! `darkTheme !== undefined`와 `darkTheme`는 무슨 차이지...? 하는 것 말이죠!

```suggestion
      if (darkTheme) {
        document.body.setAttribute("data-theme", "dark");
        return;
      }
      
      document.body.removeAttribute("data-theme");
```